lib/rmt/mirror.rb

```diff
@@ -35,14 +38,17 @@ def mirror
     replace_directory(@temp_licenses_dir, File.join(@repository_dir, '../product.license/'))
   end
 
-  def self.from_repo_model(repository, base_dir = nil)
+  def self.from_uri(uri, auth_token, repository_url: nil, base_dir: nil, to_offline: false)
```

Comment:
Technically, it works only with URLs (we don't support URNs, which are also URIs).